code/game/machinery/Sleeper.dm

```diff
@@ -216,10 +222,21 @@
 		return
 	if(occupant && occupant.reagents)
 		if(chem in injection_chems + "epinephrine")
-			if(occupant.reagents.get_reagent_amount(chem) + 10 <= 20 * efficiency)
+			var/datum/reagent/R = chemical_reagents_list[chem]
+			if(emagged == 1 && occupant.reagents.get_reagent_amount(chem) + 10 <= 20 * efficiency)

```

Comment:
Do you have a tip how to simplify those
 				occupant.reagents.add_reagent(chem, 10)
				var/units = round(occupant.reagents.get_reagent_amount(chem))
				user << "<span class='notice'>Occupant now has [units] unit\s of [chemical_reagents_list[chem]] in their bloodstream.</span>"

lines since they're all the same 3 times? I'm currently fixing the other things you mentioned.